### PR TITLE
Add OPTIONS Allowlist support to vercel terraform provider

### PR DESCRIPTION
**Pull request by kitfoster**
Add Terraform support for the new **OPTIONS Allowlist** feature

👉 [docs](https://vercel.com/docs/security/deployment-protection/methods-to-bypass-deployment-protection/options-allowlist)
👉 [changelog](https://vercel.com/changelog/options-allowlist)

### CODE REVIEW

File: vercel/data_source_project.go [hunk 1]
```diff
@@ -220,6 +220,21 @@ For more detailed information, please see the [Vercel documentation](https://ver
...
+				Computed:    true,
+				Attributes: map[string]schema.Attribute{
+					"paths": schema.ListAttribute{
+						Description: "The path to compare the request path against.",
+						Computed:    true,
+						ElementType: types.ObjectType{
+							AttrTypes: map[string]attr.Type{
+								"value": types.StringType,
+							},
+						},
```
Comment (dglsparsons) [inline vercel/data_source_project.go:234]: How come it's a list ob objects, not just a list of strings?

**Review by dglsparsons — DISMISSED** [review]

File: vercel/data_source_project.go (hunk 1 above)
Comment (kitfoster) [reply vercel/data_source_project.go:234]: Making it flexible for the future, leaves room for configurations of the paths. For example, Trusted IPs has notes. For this feature I can see supporting regex, or requiring specific headers on the request etc.

Comment (dglsparsons) [reply vercel/data_source_project.go:234]: Gotcha. Well, this LGTM 🙏

Comment (kitfoster) [reply vercel/data_source_project.go:234]: To make sure I fully answered your question, this structure is a 1-1 reflection of the db / api structure

**Review by dglsparsons — APPROVED** [review]

**Pull request merged**